src/main/java/org/jabref/gui/preferences/PreviewPrefsTab.java

```diff
@@ -212,46 +202,38 @@ public void setValues() {
             availableModel.add(Localization.lang("Preview"));
         }
 
-        btnLeft.setDisable(!chosen.getSelectionModel().getSelectedItems().isEmpty());
-        btnRight.setDisable(available.getSelectionModel().getSelectedIndices().isEmpty());
-        btnUp.setDisable(!chosen.getSelectionModel().getSelectedIndices().isEmpty());
-        btnDown.setDisable(!chosen.getSelectionModel().getSelectedIndices().isEmpty());
-
-        if (discoverCitationStyleWorker != null) {
-            discoverCitationStyleWorker.cancel(true);
+        if (citationDiscovery != null) {
+            citationDiscovery.cancel();
```

Comment:
The cancel is useless here. It just sets the task `isCanceled` to true but doesn't really abort the task. You need to implement it manually in the `call` method (e.g. in a for loop check if the task is canceled before you proceed with the next iteration). Since this is not really easy to implement here (because the cost-intensive call is living in `CitationStyle`), I would vote to use a BackgroundTask here,  make `citationDiscovery` a variable instead of a class-global field (or completely get rid of it) and remove the `cancel` part.